include/sparrow/dynamic_bitset.hpp

```diff
@@ -154,6 +178,101 @@ namespace sparrow
         dynamic_bitset_view& operator=(dynamic_bitset_view&&) = default;
     };
 
+    /**
+     *@class bitset_reference
+     */
+    template <class B>
+    class bitset_reference
+    {
+    public:
+
+        using self_type = bitset_reference<B>;
+
+        bitset_reference(const bitset_reference&) = default;
+        bitset_reference(bitset_reference&&) = default;
+
+        self_type& operator=(const self_type&) noexcept;
+        self_type& operator=(self_type&&) noexcept;
+        self_type& operator=(bool) noexcept;
+
+        explicit operator bool() const noexcept;
+
+        bool operator~() const noexcept;
+
+        self_type& operator&=(bool) noexcept;
+        self_type& operator|=(bool) noexcept;
+        self_type& operator^=(bool) noexcept;
+
+    private:
+
+        using block_type = typename B::block_type;
+        using bitset_type = B;
+
+        bitset_reference(bitset_type& bitset, block_type& block, block_type mask);
+
+        void assign(bool) noexcept;
+        void set() noexcept;
+        void reset() noexcept;
+        
+        bitset_type& m_bitset;
+        block_type& m_block;
+        block_type m_mask;
+
+        friend class bitset_iterator<B, false>;
```

Comment:
I'm not sure if it's worth associating (`friend` being the tightest bound) these types instead of just making the constructor public?

Reply:
The iterator also needs to access some private methods (for computing the blocks indices, index in a block, etc). I could make an bitset_access backdoor to limit the part of the API we expose to the iterator, but I find it a bit of overkill here.